packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -0,0 +1,253 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {APIGatewayProxyEvent, APIGatewayProxyResult} from "aws-lambda"
+import {Logger} from "@aws-lambda-powertools/logger"
+import {injectLambdaContext} from "@aws-lambda-powertools/logger/middleware"
+import {DynamoDBClient, PutItemCommand} from "@aws-sdk/client-dynamodb"
+import {marshall} from "@aws-sdk/util-dynamodb"
+import middy from "@middy/core"
+import inputOutputLogger from "@middy/input-output-logger"
+import errorHandler from "@nhs/fhir-middy-error-handler"
+import {v4 as uuidv4} from "uuid"
+
+const logger = new Logger({serviceName: "updatePrescriptionStatus"})
+const client = new DynamoDBClient({region: "eu-west-2"})
+const tableName = process.env.TABLE_NAME
+
+interface DynamoDBItem {
+  RequestID: string | undefined;
+  PrescriptionID: string;
+  PatientNHSNumber: string;
+  PharmacyODSCode: string;
+  TaskID: string;
+  LineItemID: string;
+  TerminalStatusIndicator: string;
+  LastModified: string;
+  Timestamp: string;
+  RequestMessage: any;
+}
+
+const lambdaHandler = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const xRequestId = event.headers["x-request-id"]
+    let requestBody
+    try {
+      requestBody = JSON.parse(event.body || "")
+    } catch (jsonParseError) {
+      logger.error("Error parsing JSON", {jsonParseError: jsonParseError})
+      const errorResponseBody = {
+        resourceType: "OperationOutcome",
+        issue: [
+          {
+            code: "value",
+            severity: "error",
+            details: {
+              coding: [
+                {
+                  system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
+                  code: "INVALID_JSON",
+                  display: "Invalid JSON format in request body"
+                }
```

Comment:
This needs a couple of tweaks, should probably be using the HttpErrorCode system, see: https://simplifier.net/packages/uk.nhsdigital.r4/2.8.0/files/1953636. So the coding should probably look something like:

```
coding: [
	{
		system: "https://fhir.nhs.uk/CodeSystem/http-error-codes"
		code: "BAD_REQUEST",
		display: "400: The Server was unable to process the request."
	}
]
```
I'd personally want to supplement the display to include that it was invalid json, so may be worth asking someone more familiar how strict the contents of display has to be to whats defined on the linked page.

Reply:
The error message has been changed.

Reply:
Or does the context about the invalid json belong in the diagnostic field?